crates/typst/src/geom/stroke.rs

```diff
@@ -322,6 +322,7 @@ impl Resolve for DashPattern {
 // https://tex.stackexchange.com/questions/45275/tikz-get-values-for-predefined-dash-patterns
 cast! {
     DashPattern,
+    self => dict! { "array" => self.array, "phase" => self.phase }.into_value(),
```

Comment:
I made dash pattern (returned via `.dash` field of a `stroke`) be converted to a dict with the same structure as the `DashPattern` struct. I thought of maybe comparing with the values for `"solid"`, `"dotted"` etc. to return those strings if applicable, but I settled with this in the end (it's a simpler solution, at least, and encompasses all cases). Feel free to suggest any changes.

Note that we can't directly test this conversion yet, as you can't mutate fields for now (this would be handled by a follow-up PR), and the `stroke` type only appears directly for the user through expressions like `2pt + blue` - anything more complicated (with a different dash pattern than the default `none`) would be expressed by a dict, whose values the user can retrieve directly.

Reply:
You can observe it by doing something like `#rect(stroke: (thickness: 1pt, dash: "solid")).stroke`. Looking at the current output from that, some `Debug` impl is suboptimal (it contains `Some`). I'm not super happy with the stroke setup in any case. A `stroke` constructor function instead of the dictionary would maybe make sense.

Reply:
Not returning strings like `"solid"` is the right call. The structure should be uniform and "canoncalized" to to speak. Just like `grid(columns: 2).columns == (auto, auto)`.

Reply:
> You can observe it by doing something like `#rect(stroke: (thickness: 1pt, dash: "solid")).stroke`.
Oh, didn't think of that. I'll add some tests using this. Thanks!

> Looking at the current output from that, some `Debug` impl is suboptimal (it contains `Some`).

The following lines seem to be the case:
1. Debugging `Option<DashPattern>` directly:
https://github.com/PgBiel/typst/blob/23c558fb5750dc80d01802d6e6fc0081def8642c/crates/typst/src/geom/stroke.rs#L162
2. Deriving `Debug` for `DashLength`:
https://github.com/PgBiel/typst/blob/23c558fb5750dc80d01802d6e6fc0081def8642c/crates/typst/src/geom/stroke.rs#L352

I'll take a look at those. Perhaps Typst should get its own `Debug`-like (e.g. `Repr`) trait at some point though, to avoid this kind of issue (since we can't re-implement `Debug` for `Option`, a built-in type). But that's just a thought, and perhaps a very wrong one :p

> I'm not super happy with the stroke setup in any case. A `stroke` constructor function instead of the dictionary would maybe make sense.

I agree; this could be implemented in the type rework.

Reply:
I agree on the `Repr` trait, I've actually implemented it at one point. I don't know what stopped from merging it anymore.

Reply:
Ok, fixed debugging, and added some more tests too. :+1: